cmd/mistryd/worker.go

```diff
@@ -98,10 +97,40 @@ func (s *Server) Work(ctx context.Context, j *Job) (buildInfo *types.BuildInfo,
 	}
 
 	log.Printf("Creating new build directory...")
-	cleanupPending, err := j.BootstrapBuildDir(s.cfg.FileSystem, log)
+	err = j.BootstrapBuildDir(s.cfg.FileSystem, log)
+	if err != nil {
+		err = workErr("could not bootstrap build dir", err)
+	}
 
+	// moves the pending directory to the ready one
 	defer func() {
-		if cleanupPending {
+		rerr := os.Rename(j.PendingBuildPath, j.ReadyBuildPath)
+		if rerr != nil {
+			errstr := "could not move pending path"
+			if err == nil {
+				err = fmt.Errorf("%s; %s", errstr, rerr)
+			} else {
+				err = fmt.Errorf("%s; %s | %s", errstr, rerr, err)
+			}
+		}
+
+		// When there are no errors the LatestBuildPath can be updated
+		// to point to the last build. Otherwise it remains unchanged.
+		if err == nil {
+			_, err = os.Lstat(j.LatestBuildPath)
```

Comment:
What happens if the `err` from `os.Lstat` is not nil?

Reply:
Nice catch.